src/extension.ts

```diff
@@ -137,17 +138,44 @@ export function activate(context: vs.ExtensionContext) {
 	// TODO: Check if EventEmitter<T> would be more appropriate than our own.
 
 	// Set up providers.
-	context.subscriptions.push(vs.languages.registerHoverProvider(DART_MODE, new DartHoverProvider(analyzer)));
-	context.subscriptions.push(vs.languages.registerDocumentFormattingEditProvider(DART_MODE, new DartFormattingEditProvider(analyzer)));
-	context.subscriptions.push(vs.languages.registerOnTypeFormattingEditProvider(DART_MODE, new DartTypeFormattingEditProvider(analyzer), "}", ";"));
-	context.subscriptions.push(vs.languages.registerCompletionItemProvider(DART_MODE, new DartCompletionItemProvider(analyzer), ".", ":", " ", "=", "("));
-	context.subscriptions.push(vs.languages.registerDefinitionProvider(DART_MODE, new DartDefinitionProvider(analyzer)));
-	context.subscriptions.push(vs.languages.registerDocumentSymbolProvider(DART_MODE, new DartDocumentSymbolProvider(analyzer)));
-	context.subscriptions.push(vs.languages.registerReferenceProvider(DART_MODE, new DartReferenceProvider(analyzer)));
+	let hoverProvider = new DartHoverProvider(analyzer);
+	let formattingEditProvider = new DartFormattingEditProvider(analyzer);
+	let typeFormattingEditProvider = new DartTypeFormattingEditProvider(analyzer);
+	let completionItemProvider = new DartCompletionItemProvider(analyzer);
+	let definitionProvider = new DartDefinitionProvider(analyzer);
+	let documentSymbolProvider = new DartDocumentSymbolProvider(analyzer);
+	let referenceProvider = new DartReferenceProvider(analyzer);
+	let documentHighlightProvider = new DartDocumentHighlightProvider(analyzer);
+	let codeActionProvider = new DartCodeActionProvider(analyzer);
+	let renameProvider = new DartRenameProvider(analyzer);
+
+	context.subscriptions.push(vs.languages.registerHoverProvider(DART_MODE, hoverProvider));
+	context.subscriptions.push(vs.languages.registerDocumentFormattingEditProvider(DART_MODE, formattingEditProvider));
+	context.subscriptions.push(vs.languages.registerOnTypeFormattingEditProvider(DART_MODE, typeFormattingEditProvider, "}", ";"));
+	context.subscriptions.push(vs.languages.registerCompletionItemProvider(DART_MODE, completionItemProvider, ".", ":", " ", "=", "("));
+	context.subscriptions.push(vs.languages.registerDefinitionProvider(DART_MODE, definitionProvider));
+	context.subscriptions.push(vs.languages.registerDocumentSymbolProvider(DART_MODE, documentSymbolProvider));
+	context.subscriptions.push(vs.languages.registerReferenceProvider(DART_MODE, referenceProvider));
+	context.subscriptions.push(vs.languages.registerDocumentHighlightProvider(DART_MODE, documentHighlightProvider));
+	context.subscriptions.push(vs.languages.registerCodeActionsProvider(DART_MODE, codeActionProvider));
+	context.subscriptions.push(vs.languages.registerRenameProvider(DART_MODE, renameProvider));
+
+	// Analyze Angular2 templates, requires the angular_analyzer_plugin.
+	if (config.analyzeAngularTemplates) {
```

Comment:
Changed to using an array of filters and forEach-ing. I left registerWorkspaceSymbolProvider as is.The only one handled manually is registerOnTypeFormattingEditProvider, since this is no corresponding implementation in the angular_analysis_plugin